### ISSUE TITLE
Project Rotation: repometrics.json

### ISSUE THREAD

**Issue by decause-gov** [DSACMS/repo-scaffolder#97]
## MENTOR
- @natalialuzuriaga 
- SWE USDC 2023
- Communication Channel(s) for your team/project (e.g. Slack Channel URL, GitHub Repo URL, etc...)

## BRIEF DESCRIPTION
- source a list of user generated metrics/labels to include in repometrics.json 
- create a repometrics.json file that users can edit manually
- create a cookiecutter.json file that edits repometrics.json programmatically

## SCOPING
- Days

## SKILLS NEEDED
e.g. Back-end, Webdev

## DESIRED DELIVERY WINDOW
- June
- July
- August
- Any

## NOTES/INSPO

- formatted metrics.json inspo: https://github.com/DSACMS/metrics/tree/main/app/site/_data/DSACMS/metrics
- cookiecutter inspo (link)

# ACKNOWLEDGEMENTS
- [X] - Yes, I agree to spend file a project proposal ticket
- [ ] - Yes, I agree this project falls within a 2 week or less (ideally less) scope of time
- [ ] - Yes, I agree to spend an hour per week of office hours and 1x1 mentoring
- [ ] - Yes, I agree to spend an hour at the end of the sprint for filling out a Retro Template

**Comment by decause-gov** [comment]
06/17:
- Met with Nat. Discussed data for inputs and tags
- planning to meet again tomorrow to finalize as a team
- will go over different libs
- goal: have all repometrics identified for the .json file
- goal: ideas/mockups for a visualization.
- goal: simple text in the repo reports

**Comment by decause-gov** [comment]
06/21:
- Determined fields for repometrics.json
- PR incoming for template on repo-scaffolder 
- complete: repometrics identified 
- Next Week: mockups
- future: text in repo reports

**Comment by decause-gov** [comment]
06/24:
- Bulk of the code exists, will change templates, and then move on to creating mockups.
- Goal: finish templates, create mocks.
- Stetch: Start implementing in reports!

**Comment by RicardoZamora01** [comment]
**Note**: To have nested cookiecutters, the parent cookiecutter.json must have the flag:

`"_copy_without_render": [<subdirectory>]`

this will copy the subdirectory and the contents (child cookiecutter) without invoking it.

https://stackoverflow.com/questions/76262434/how-do-i-create-a-nested-cookiecutter

**Comment by IsaacMilarky** [comment]
- Good work on the tier 2 repometrics https://github.com/DSACMS/repo-scaffolder/pull/106
- ACTION: replace Python script with Bash script before PR merge
- Not going to be creating a repometrics.json file, instead going to use the code.json file 👍

**Comment by IsaacMilarky** [comment]
7/1:

- Merged tier 2, need to do tier 3 and 4
- Need to figure out how to get data on metrics site
- ACTION: Create frontend ticket for repometrics on the metrics website 
- ACTION: Create backend ticket for repometrics on the metrics website 
- Get rest of the tiers merged first

**Comment by RicardoZamora01** [comment]
7/3:

- all tiers PR's merged!
- ACTION: User research plan - Get feedback from around CMS with forming the categories/fields, update code with new fields if needed

**Comment by IsaacMilarky** [comment]
7/8:

- Follow up on User Research Plan. See if we want to add fields or not. 
- Close after action item completed

**Comment by IsaacMilarky** [comment]
7/12:

- Still setting up meetings and working on User Research

**Comment by natalialuzuriaga** [comment]
7/15:
- Backend is all done 🙌  waiting on Isaac's review @IsaacMilarky 
- Next time site will update is next Monday so get in code.json added to metrics, DedupliFHIR, /open, repo-scaffolder

**Comment by natalialuzuriaga** [comment]
7/19:
- Frontend PR is up with 6 tags up! Needs to double check with @natalialuzuriaga on editing some options (repo host)
- For later: need to add required code.json fields to repo-scaffolder, user research on hold for now
- code.json has been added to 4 DSACMS repos

**Comment by decause-gov** [comment]
7/22:
- Works in dev, not in build, need to be fixed (path issue)
- Once merged, close this ticket!

**Comment by decause-gov** [comment]
07/26:
- Path issues resolved!
- PR Merged! https://github.com/DSACMS/metrics/pull/189
